jmespath-core/src/main/java/io/burt/jmespath/function/CompareByFunction.java

```
@@ -0,0 +1,56 @@
+package io.burt.jmespath.function;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+
+import io.burt.jmespath.Adapter;
+import io.burt.jmespath.node.JmesPathNode;
+
+public class CompareByFunction extends JmesPathFunction {
+  private final int compareModifier;
+
+  public CompareByFunction(int compareModifier) {
+    super(2, 2);
+    this.compareModifier = compareModifier;
+  }
+
+  @Override
+  protected <T> T internalCall(Adapter<T> adapter, List<ExpressionOrValue<T>> arguments) {
+    T array = arguments.get(0).value();
+    JmesPathNode expression = arguments.get(1).expression();
+    if (arguments.get(0).isExpression()) {
```

I want to get the name of the function into the error message, and that's not available in `ExpressionOrValue`. I guess I could catch all exceptions in `JmesPathFunction` and add the name, but not sure if I want to.